core/node/node_framework/src/implementations/layers/da_dispatcher.rs

```diff
@@ -0,0 +1,77 @@
+use zksync_config::{
+    configs::{da_dispatcher::DADispatcherConfig, eth_sender::PubdataSendingMode},
+    EthConfig,
+};
+use zksync_da_layers::DataAvailabilityInterface;
+use zksync_dal::Core;
+use zksync_db_connection::connection_pool::ConnectionPool;
+
+use crate::{
+    implementations::resources::pools::{MasterPool, PoolResource},
+    service::{ServiceContext, StopReceiver},
+    task::Task,
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+#[derive(Debug)]
+pub struct DataAvailabilityDispatcherLayer {
+    da_config: DADispatcherConfig,
+    eth_config: EthConfig,
+}
+
+impl DataAvailabilityDispatcherLayer {
+    pub fn new(da_config: DADispatcherConfig, eth_config: EthConfig) -> Self {
+        Self {
+            da_config,
+            eth_config,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl WiringLayer for DataAvailabilityDispatcherLayer {
+    fn layer_name(&self) -> &'static str {
+        "da_dispatcher_layer"
+    }
+
+    async fn wire(self: Box<Self>, mut context: ServiceContext<'_>) -> Result<(), WiringError> {
+        let master_pool_resource = context.get_resource::<PoolResource<MasterPool>>().await?;
+        let master_pool = master_pool_resource.get().await.unwrap();
+
+        if self.eth_config.sender.unwrap().pubdata_sending_mode == PubdataSendingMode::Custom {
+            let da_client = zksync_da_client::new_da_client(self.da_config.clone()).await;
```

Comment:
What async operations can be performed here? If they can take a long time, it may make sense to postpone initialization so that it's encapsulated in `DataAvailabilityDispatcherTask`.

Reply:
removed that func, so not relevant anymore, but initially it was done this way because `ObjectStoreFactory::create_from_config` is async